week10/Week10.playground/Contents.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+struct CatFact: Decodable {
+    let fact: String
+    let length: Int
+}
+
+struct CatFactSequence: AsyncSequence {
+    typealias Element = CatFact
+
+    let numberOfFacts: Int
+
+    func makeAsyncIterator() -> Iterator {
+        Iterator(numberOfFacts: numberOfFacts)
+    }
+
+    struct Iterator: AsyncIteratorProtocol {
+        let numberOfFacts: Int
```

Comment:
A small suggestion if we want to stick to the strict `Iterator` design pattern: it would be better to omit the limit from it. In this case, I think it would be better to put in a `fetchCatFacts(numberOfFacts:)` function. This way, we also don't have to limit ourselves up front in cases when we're looking for some specific data.